### PR TITLE
Create Cylc Header component

### LINKED ISSUE #120: Create Cylc header component
For #116, in the left top corner, what we have right now needs to be replaced by this new component.

image

Now it must contain the Cylc logo, and the user name and host name.

image

### PR DESCRIPTION
**Pull request by kinow**
Add a new `Header` Cylc component. Closes #120 

Screenshot (this issue is just the header, the left top part in the sidebar, but changed the style of the sidebar as well to reflect better what changed :+1: a future PR should address the rest of the sidebar):

Screen Shot 2019-07-15 at 20 02 03-fullpage

(tried to run the linter with the Standard JS, to minimize conflicts if #125 goes in)

### CODE REVIEW

**Comment by kinow** [comment]
This component has two props, `user` and `environment`, both with default values _guest_ and _development_ respectively (as this way it will be easier to test without the backend, but later we should decide where to grab this information from and update the component)

File: src/components/cylc/Header.vue [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      type: String,
+      default: 'guest'
+    },
+    environment: {
+      type: String,
+      default: 'development'
+    }
+  },
+  data: () => ({
+    logo: './img/logo.png'
```
Comment (oliver-sanders) [inline src/components/cylc/Header.vue]: Might as well use SVG.

Comment (kinow) [reply src/components/cylc/Header.vue]: Oh, good idea. Only pending a quick question on whether someone will want to customize it.

**Comment by kinow** [comment]
Exported original SVG as optimized SVG, good idea suggested by @oliver-sanders 

```xml
<svg version="1.1" viewBox="0 0 655 260" xmlns="http://www.w3.org/2000/svg" xmlns:osb="http://www.openswatchbook.org/uri/2009/osb">
 <g transform="translate(292.53 -49.505)">
  <g>
   <circle transform="scale(-1,1)" cx="-135.7" cy="248.39" r="27.743" fill="#818181"/>
   <circle cx="201.05" cy="216.65" r="72.857" fill="#676767" fill-rule="evenodd"/>
   <circle cx="167.35" cy="136.32" r="37.379" fill="#888"/>
   <circle cx="292.32" cy="121.63" r="49.8" fill="#a4a4a4"/>
   <g fill="#4a4a4a">
    <path d="m-239.15 150.44q5.28 0 17.92 1.92l3.84 0.48-0.48 9.76q-12.8-1.44-18.88-1.44-13.6 0-18.56 6.56-4.8 6.4-4.8 24 0 17.44 4.48 24.32 4.64 6.88 19.04 6.88l18.88-1.44 0.48 9.92q-14.88 2.24-22.24 2.24-18.72 0-25.92-9.6-7.04-9.6-7.04-32.32 0-22.88 7.68-32 7.68-9.28 25.6-9.28z"/>
    <path d="m-175.59 152.04h12l20 69.6h5.28l20.16-69.6h12l-33.28 115.52h-12l10.4-35.52h-11.84z"/>
    <path d="m-58.613 232.04v-114.88h12v114.88z"/>
    <path d="m34.535 150.44q5.28 0 17.92 1.92l3.84 0.48-0.48 9.76q-12.8-1.44-18.88-1.44-13.6 0-18.56 6.56-4.8 6.4-4.8 24 0 17.44 4.48 24.32 4.64 6.88 19.04 6.88l18.88-1.44 0.48 9.92q-14.88 2.24-22.24 2.24-18.72 0-25.92-9.6-7.04-9.6-7.04-32.32 0-22.88 7.68-32 7.68-9.28 25.6-9.28z"/>
   </g>
   <text x="-431.42239" y="80.933609" fill="#000000" font-family="'Titillium Web'" font-size="53.333px" font-weight="bold" letter-spacing="0px" word-spacing="0px" style="line-height:1.25" xml:space="preserve"><tspan x="-431.42239" y="127.32852"/></text>
  </g>
 </g>
</svg>
```

@cylc/core just checking because some software allow it: are we going to allow users to change the Cylc logo in Cylc UI? If so we may need to leave the PNG. Otherwise we can switch to SVG.

**Comment by oliver-sanders** [comment]
> are we going to allow users to change the Cylc logo in Cylc UI

I hope not!

**Comment by kinow** [comment]
Pinged riot to see what others think :+1:

**Comment by hjoliver** [comment]
I agree with @oliver-sanders - I can't think of any reason why users should be able to change the Cylc logo!

**Comment by kinow** [comment]
In that case, updated PR replacing image based logo by the SVG.

image

File: src/components/cylc/Header.vue [hunk 2]
```diff
@@ -0,0 +1,46 @@
+<template>
+  <v-list-tile-action-text class="pt-3 pb-3 c-header">
+    <v-layout align-center justify-center row wrap fill-height>
+      <div class="mb-2">
+        <svg version="1.1" viewBox="0 0 655 260" xmlns="http://www.w3.org/2000/svg" xmlns:osb="http://www.openswatchbook.org/uri/2009/osb">
+          <g transform="translate(292.53 -49.505)">
```
Comment (oliver-sanders) [inline src/components/cylc/Header.vue]: [inkscape rant]

Gotta love Inkscape, hey lets define a document 655x260 but then wrap everything using a completely unecessary and totally arbitrary transform of 292.53x-49.505, that'll make all the number really nice!

[/inkscape rant]

Good job exporting with 5sf precision though!

File: src/components/cylc/Header.vue [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+              <circle cx="292.32" cy="121.63" r="49.8" fill="#a4a4a4"/>
+              <g fill="#4a4a4a">
+                <path d="m-239.15 150.44q5.28 0 17.92 1.92l3.84 0.48-0.48 9.76q-12.8-1.44-18.88-1.44-13.6 0-18.56 6.56-4.8 6.4-4.8 24 0 17.44 4.48 24.32 4.64 6.88 19.04 6.88l18.88-1.44 0.48 9.92q-14.88 2.24-22.24 2.24-18.72 0-25.92-9.6-7.04-9.6-7.04-32.32 0-22.88 7.68-32 7.68-9.28 25.6-9.28z"/>
+                <path d="m-175.59 152.04h12l20 69.6h5.28l20.16-69.6h12l-33.28 115.52h-12l10.4-35.52h-11.84z"/>
+                <path d="m-58.613 232.04v-114.88h12v114.88z"/>
+                <path d="m34.535 150.44q5.28 0 17.92 1.92l3.84 0.48-0.48 9.76q-12.8-1.44-18.88-1.44-13.6 0-18.56 6.56-4.8 6.4-4.8 24 0 17.44 4.48 24.32 4.64 6.88 19.04 6.88l18.88-1.44 0.48 9.92q-14.88 2.24-22.24 2.24-18.72 0-25.92-9.6-7.04-9.6-7.04-32.32 0-22.88 7.68-32 7.68-9.28 25.6-9.28z"/>
+              </g>
+              <text x="-431.42239" y="80.933609" fill="#000000" font-family="'Titillium Web'" font-size="53.333px" font-weight="bold" letter-spacing="0px" word-spacing="0px" style="line-height:1.25" xml:space="preserve"><tspan x="-431.42239" y="127.32852"/></text>
```
Comment (oliver-sanders) [inline src/components/cylc/Header.vue]: I have no idea why Inkscape put this `<text>` node here as you've included the font as an SVG path avoiding the need to import a web font (+1 BTW).

I think we can just remove this dud `<text>` node?

File: src/components/cylc/Header.vue (hunk 2 above)
Comment (kinow) [reply src/components/cylc/Header.vue]: First time I did it for the web, just followed a tutorial. :relieved:

Comment (oliver-sanders) [reply src/components/cylc/Header.vue]: Nothing you've done, Just Inkscape being daft!

File: src/components/cylc/Header.vue (hunk 3 above)
Comment (kinow) [reply src/components/cylc/Header.vue]: Oh, good spot. Removed :+1:

Comment (kinow) [reply src/components/cylc/Header.vue]: (and tested and the logo is still looking good)

Comment (oliver-sanders) [reply src/components/cylc/Header.vue]: Cool, saves the browser poking around trying to find Titillium Web!

**Comment by wxtim** [comment]
Very nice. Looks pretty smart on a simulated Galaxy S5 too. I'm ready to approve this and merge it.
Screen Shot 2019-07-24 at 09 45 58

**Pull request merged**